### PR TITLE
Add initial Torus wallet integration

### PR DESCRIPTION
**Pull request by adotcruz**
This PR adds Torus as the Web3 provider.

This CL also breaks up wallet logic into 3 steps:

1. Torus api is initialized for application to use.
2. User logs in and authorizes for use of Torus wallet
3. After a user is logged in, Web3 is initialized for the application.

I also added react-bootstrap + bootstrap for use of components.

I had to `npm i graphql` because it is a peer dependency for some of the other graphql node modules we use.

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -25,11 +30,14 @@ class App extends React.Component<Record<string, unknown>, AppState> {
...
+
+    // Need to bind to `this` since this function is called from the DOM, where this loses its scope.
+    this.signUserIn = this.signUserIn.bind(this);
   }
 
   private async getAccountInfo() {
-    this.web3 = await getWeb3();
-
+    // Must pass in Torus provider in order for Web3 to not use MetaMask by default.
+    this.web3 = new Web3(this.state.torus.provider);
```
Comment (jw122) [inline src/components/App.tsx:40]: This makes sense. Wonder if it also gets passed in later as `currentProvider` when we call  `deposit(this.web3.eth.currentProvider, accounts[0]);` in L53

File: src/web3/wallet.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+// The configuration passed into `torus.init` is what determines which network Torus is on, as well
+// as other block chain information
+export const initializeTorusConnection: () => Promise<Torus> = async (): Promise<Torus> => {
+  const torus = new Torus({
+    buttonPosition: "top-left", // default: bottom-left
+  });
+  await torus.init({
+    buildEnv: "production", // default: production
+    enableLogging: true, // default: false
+    network: {
```
Comment (jw122) [inline src/web3/wallet.ts:15]: so this is already on Polygon's testnet? that's cool

**Review by jw122 — APPROVED** [review]
This is awesome Anthony! I like how the comments keep things clear/simple. Just left a couple questions.

Also we shouldn't have to check in the package-lock.json file -- not sure why .gitignore isn't working. Maybe this change was pushed before you pulled the .gitignore I added?

File: src/components/App.tsx (hunk 1 above)
Comment (adotcruz) [reply src/components/App.tsx:40]: yeah I think it's the similar call as I assume `this.web3.eth.currentProvider == this.state.torus.provider`

File: src/web3/wallet.ts (hunk 2 above)
Comment (adotcruz) [reply src/web3/wallet.ts:15]: I think so(?) I got it directly from the docs

**Pull request merged**